servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java

```diff
@@ -27,7 +28,7 @@
  * A greedy allocator that returns the first Storage dir fitting the size of block to allocate.
  * This class serves as an example how to implement an allocator.
  */
-public class GreedyAllocator implements Allocator {
+public class GreedyAllocator extends BlockStoreEventListenerBase implements Allocator {
```

Comment:
Maybe it makes sense to make Evictor/Allocator extend the `BlockStoreEventListenerBase`? Its kind of weird to make all the evictors need to extend `BlockStoreEventListenerBase`.